pkg/issuer/acme/dns/azuredns/azuredns.go

```diff
@@ -150,7 +186,8 @@ func getAuthorization(env azure.Environment, clientID, clientSecret, subscriptio
 			// RefreshToken is absent from responses.
 			err = newSPT.Refresh()
 			if err != nil {
-				return nil, err
+				logf.Log.V(logf.ErrorLevel).Error(err, "failed to refresh token")
```

Comment:
I like that you log the full original error  to aid debugging:


> E0103 15:55:09.853144       1 azuredns.go:189] cert-manager "msg"="failed to refresh token" "error"="adal: Refresh request failed. Status Code = '400'. Response body: {\"error\":\"invalid_request\",\"error_description\":\"AADSTS70021: No matching federated identity record found for presented assertion. Assertion Issuer: 'https://eastus2.oic.prod-aks.azure.com/f99bd6a4-665c-41cf-aff1-87a89d5c62d4/b64ce3fe-190e-4c88-9266-2d1326f58dab/'. Assertion Subject: 'system:serviceaccount:cert-manager:cert-manager'. Assertion Audience: 'api://AzureADTokenExchange'. https://docs.microsoft.com/en-us/azure/active-directory/develop/workload-identity-federation\\r\\nTrace ID: 75e2e33f-61dd-402b-a735-d4aad1126c00\\r\\nCorrelation ID: 36106c6b-192f-4add-b0e1-6dcbf2c91a1f\\r\\nTimestamp: 2023-01-03 15:55:09Z\",\"error_codes\":[70021],\"timestamp\":\"2023-01-03 15:55:09Z\",\"trace_id\":\"75e2e33f-61dd-402b-a735-d4aad1126c00\",\"correlation_id\":\"36106c6b-192f-4add-b0e1-6dcbf2c91a1f\",\"error_uri\":\"https://login.microsoftonline.com/error?code=70021\"} Endpoint https://login.microsoftonline.com/f99bd6a4-665c-41cf-aff1-87a89d5c62d4/oauth2/token?api-version=1.0" 

And I observed that this error is now logged much frequently, since the challenge.status.reason field is now stable so the challenge is not continually reconciled.